src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -16565,94 +16582,115 @@ def dt_round(
-
-        # We need to implement the algorithm for rounding half to even whenever
-        # the date value is at half point of the slice:
-        # https://en.wikipedia.org/wiki/Rounding#Rounding_half_to_even
-
-        # First, we need to calculate the length of half a slice.
-        # This is straightforward if the length is already even.
-        # If not, we then need to first downlevel the freq to a
-        # lower granularity to ensure that it is even.
+        dtype = self.index_dtypes[0] if include_index else self.dtypes[0]
+        if is_datetime64_any_dtype(dtype):
+            slice_length, slice_unit = rule_to_snowflake_width_and_slice_unit(
+                rule=freq  # type:  ignore[arg-type]
+            )
 
-        def down_level_freq(slice_length: int, slice_unit: str) -> tuple[int, str]:
-            if slice_unit == "minute":
-                slice_length *= 60
-                slice_unit = "second"
-            elif slice_unit == "hour":
-                slice_length *= 60
-                slice_unit = "minute"
-            elif slice_unit == "day":
-                slice_length *= 24
-                slice_unit = "hour"
-            else:
-                f"Snowpark pandas 'Series.dt.round' method doesn't support setting 'freq' parameter with '{slice_unit}' unit"
-            return slice_length, slice_unit
+            if (
+                slice_unit not in SUPPORTED_DT_FLOOR_CEIL_FREQS
+                or slice_unit == "second"
+            ):
+                ErrorMessage.parameter_not_implemented_error(
+                    f"freq={freq}", method_name
+                )
 
-        if slice_length % 2 == 1:
-            slice_length, slice_unit = down_level_freq(slice_length, slice_unit)
-        half_slice_length = int(slice_length / 2)
+            # We need to implement the algorithm for rounding half to even whenever
+            # the date value is at half point of the slice:
+            # https://en.wikipedia.org/wiki/Rounding#Rounding_half_to_even
+
+            # First, we need to calculate the length of half a slice.
+            # This is straightforward if the length is already even.
+            # If not, we then need to first downlevel the freq to a
+            # lower granularity to ensure that it is even.
+
+            def down_level_freq(slice_length: int, slice_unit: str) -> tuple[int, str]:
+                if slice_unit == "minute":
+                    slice_length *= 60
+                    slice_unit = "second"
+                elif slice_unit == "hour":
+                    slice_length *= 60
+                    slice_unit = "minute"
+                elif slice_unit == "day":
+                    slice_length *= 24
+                    slice_unit = "hour"
+                else:
+                    f"Snowpark pandas 'Series.dt.round' method doesn't support setting 'freq' parameter with '{slice_unit}' unit"
```

Comment:
raise error?

Reply:
This is original code I moved inside if block. @sfc-gh-helmeleegy is this intentional?